### PR TITLE
Added Inputs for text and image

### PR DESCRIPTION
**Pull request by TomerGibor**

### CODE REVIEW

File: Dev/VideoChatClient/Input/IInput.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#include <cstdint>
```
Comment (TomerGibor) [inline Dev/VideoChatClient/Input/IInput.hpp:1]: #pragma once

File: Dev/VideoChatClient/Input/TextInput.hpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include <iostream>
```
Comment (TomerGibor) [inline Dev/VideoChatClient/Input/TextInput.hpp:1]: #pragma once

File: Dev/VideoChatClient/Input/IInput.hpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+typedef std::vector<uint8_t> Buffer;
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/IInput.hpp:6]: Create a Common.hpp file and put this there, change in all files.

Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/IInput.hpp:6]: use Using instead of typedef, race the Resharper!!🐱‍🏍

File: Dev/VideoChatClient/Input/IInput.hpp [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+class IInput
+{
+public:
+	IInput() = default;
+	virtual ~IInput() = default;
+
+	virtual void take_input() = 0;
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/IInput.hpp:14]: Great

File: Dev/VideoChatClient/Input/ImageInput.cpp [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+#include "VideoChatClientException/VideoChatClientException.hpp"
+
+ImageInput::ImageInput() : m_capture()
+{
+	m_capture.open(cv::CAP_ANY);
+	if (!m_capture.isOpened())
+	{
+		throw VideoChatClientException(VideoChatClientStatus::IMAGE_INPUT_OPEN_CAPTURE_FAILED, 0); //TODO
+	}
+	m_capture.set(cv::CAP_PROP_FRAME_WIDTH, IMAGE_WIDTH);
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/ImageInput.cpp:12]: I see you made the width&height as consts, maybe we would like later to get the screen or windows resolution and change according to that. just a note...

File: Dev/VideoChatClient/Input/ImageInput.cpp [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+{
+	cv::Mat frame{};
+
+	m_capture.read(frame);
+
+	if (frame.empty())
+	{
+		throw VideoChatClientException(VideoChatClientStatus::IMAGE_INPUT_READ_FRAME_FAILED, 0); //TODO
+	}
+	cv::imencode(".jpg", frame, m_data, std::vector<int32_t>());
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/ImageInput.cpp:26]: I don't think that passing "std::vector<int32_t>()" is what you should pass.
it should be used for extra flags like the quality of the image and you are passing nothing...
read the docs of it👀

Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/ImageInput.cpp:26]: and this function is great, good job for finding that

File: Dev/VideoChatClient/Main.cpp [hunk 7]
```diff
@@ -2,13 +2,13 @@
...
 #include "AutoInitWinSock/AutoInitWinSock.hpp"
+#include "Input/ImageInput.hpp"
 #include "Socket/Socket.hpp"
 
-
-#include "opencv2/opencv.hpp"
-
 int wmain()
 {
+	ImageInput im{};
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Main.cpp:10]: We should think how we want to use the input/output generically, that we can send image and after that voice while keeping the right expected flow

File: Dev/VideoChatClient/Input/IInput.hpp [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+#include <cstdint>
+#include <vector>
+
+typedef std::vector<uint8_t> Buffer;
+
+class IInput
+{
+public:
+	IInput() = default;
+	virtual ~IInput() = default;
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/IInput.hpp:12]: Add deletion of the copy C'tor and operator=

File: Dev/VideoChatClient/Input/ImageInput.cpp [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+void ImageInput::take_input()
+{
+	cv::Mat frame{};
+
+	m_capture.read(frame);
+
+	if (frame.empty())
+	{
+		throw VideoChatClientException(VideoChatClientStatus::IMAGE_INPUT_READ_FRAME_FAILED, 0); //TODO
```
Comment (ron-elmekiesse) [inline Dev/VideoChatClient/Input/ImageInput.cpp:24]: remove comment

File: Dev/VideoChatClient/Input/ImageInput.cpp (hunk 6 above)
Comment (TomerGibor) [reply Dev/VideoChatClient/Input/ImageInput.cpp:26]: Actually, that's how you're supposed to do it. Just saw that it also has a default value
image

File: Dev/VideoChatClient/Input/IInput.hpp (hunk 8 above)
Comment (TomerGibor) [reply Dev/VideoChatClient/Input/IInput.hpp:12]: Why?

File: Dev/VideoChatClient/Input/ImageInput.cpp (hunk 9 above)
Comment (TomerGibor) [reply Dev/VideoChatClient/Input/ImageInput.cpp:24]: If not a win error, should I pass 0?

File: Dev/VideoChatClient/Main.cpp (hunk 7 above)
Comment (TomerGibor) [reply Dev/VideoChatClient/Main.cpp:10]: Thought about it, maybe double Inheritance? Or another class for AV

File: Dev/VideoChatClient/Input/IInput.hpp (hunk 3 above)
Comment (TomerGibor) [reply Dev/VideoChatClient/Input/IInput.hpp:6]: Can't! I tried 😥

Comment (TomerGibor) [reply Dev/VideoChatClient/Input/IInput.hpp:6]: How would I create it in both branches simultaneously?

**Pull request merged**